contrib/babelfishpg_common/sql/geography.sql

```diff
@@ -232,27 +336,47 @@ CREATE OR REPLACE FUNCTION sys.Geography__STPointFromText(text, integer)
 		lat float8;
 	BEGIN
 		srid := $2;
-		-- Here we're flipping the coordinates since Geography Datatype stores the point from STGeomFromText and STPointFromText in Reverse Order i.e. (long, lat) 
-		lat = (SELECT sys.lat(sys.Geography__STFlipCoordinates(sys.stgeogfromtext_helper($1, $2))));
-		IF srid = ANY(valid_srids) AND lat >= -90.0 AND lat <= 90.0 THEN
-			-- Call the underlying function after preprocessing
-			-- Here we're flipping the coordinates since Geography Datatype stores the point from STGeomFromText and STPointFromText in Reverse Order i.e. (long, lat) 
-			RETURN (SELECT sys.Geography__STFlipCoordinates(sys.stgeogfromtext_helper($1, $2)));
-		ELSEIF lat < -90.0 OR lat > 90.0 THEN
-			RAISE EXCEPTION 'Latitude values must be between -90 and 90 degrees';
+		-- Here we are flipping the coordinates since Geography Datatype stores the point from STGeomFromText and STPointFromText in Reverse Order i.e. (long, lat) 
+		geom = (SELECT sys.stgeogfromtext_helper($1, $2));
+		Geomtype = (SELECT sys.ST_GeometryType(geom));
+		IF Geomtype = 'ST_Point' THEN
+			lat = (SELECT sys.lat(sys.Geography__STFlipCoordinates(sys.stgeogfromtext_helper($1, $2))));
+			IF srid = ANY(valid_srids) AND lat >= -90.0 AND lat <= 90.0 THEN
+				-- Call the underlying function after preprocessing
+				-- Here we are flipping the coordinates since Geography Datatype stores the point from STGeomFromText and STPointFromText in Reverse Order i.e. (long, lat) 
+				IF (SELECT sys.ST_Zmflag(geom)) = 1 OR (SELECT sys.ST_Zmflag(geom)) = 2 OR (SELECT sys.ST_Zmflag(geom)) = 3 THEN
+					RAISE EXCEPTION 'Z and M flags are currently not supported in Babelfish';
+				ELSE
+					RETURN (SELECT sys.Geography__STFlipCoordinates(geom));
+				END IF;
+			ELSEIF lat < -90.0 OR lat > 90.0 THEN
+				RAISE EXCEPTION 'Latitude values must be between -90 and 90 degrees';
+			ELSE
+				RAISE EXCEPTION 'Inavalid SRID';
+			END IF;
 		ELSE
-			RAISE EXCEPTION 'Inavalid SRID';
+			RAISE EXCEPTION '% is currently not supported in Babelfish', Geomtype;
```

Comment:
Could we be more specific, say '% is currently not a supported SRID in Babelfish'? Also, is it custom to use the word Babelfish in error messages? Or is it more idiomatic to just say '% is currently not a supported SRID.'

Reply:
Same for all the places this exception (and similar ones) is raised

Reply:
Here, **'Geomtype'** is geometry type(such as Point, LineString, etc) and not SRID, so it makes sense for throwing an error message like ```LINESTRING is currently not supported``` 

Although removed babelfish from the error messages. Now error messages look like '% is currently not a supported'